File: core/src/main/java/javabot/operations/OperationComparator.java
package javabot.operations;

import java.util.Comparator;

public class OperationComparator implements Comparator<BotOperation> {
    @Override
    public int compare(final BotOperation botOperation, final BotOperation o) {
        int value=Integer.compare(o.getPriority(), botOperation.getPriority());
        if(value==0) {
            value = botOperation.getName().compareTo(o.getName());
        }
        return value;
    }
}

File: src/main/kotlin/javabot/operations/OperationComparator.kt
package javabot.operations

import java.util.Comparator

class OperationComparator : Comparator<BotOperation> {
    override fun compare(botOperation: BotOperation, o: BotOperation): Int {
        var value = Integer.compare(o.getPriority(), botOperation.getPriority())
        if (value == 0) {
            value = botOperation.getName().compareTo(o.getName())
        }
        return value
    }
}